File: level2/장애물인식프로그램.cpp
#include<bits/stdc++.h>
using namespace std;

int dx[] = {-1, 1, 0, 0};
int dy[] = {0, 0, -1, 1};

int N;
char mapp[26][26];
int map_flag[26][26];
vector <int> ans;

int bfs(int x, int y){
    int cnt = 1;
    queue<pair<int, int> > q;
    q.push({x, y});
    map_flag[x][y] = 1;
    while (!q.empty()){
        int x = q.front().first;
        int y = q.front().second;
        q.pop();
        for (int i=0; i<4; i++){
            int nx = x+dx[i]; int ny = y+dy[i];
            if (nx>-1 and nx<N and ny>-1 and ny<N){
                if(mapp[nx][ny] == '1' and map_flag[nx][ny]==0){
                    q.push({nx,ny});
                    map_flag[nx][ny] = 1;
                    cnt+=1;
                }
            }
        }
    }
    return cnt;
}

int main(int argc, char** argv)
{
    cin >> N;
    for (int i=0; i<N; i++){
        for (int j=0; j<N;j++){
            char ch;
            cin >> ch;
            mapp[i][j] = ch;
        }
    }
    for (int i=0; i<N; i++){
        for (int j=0; j<N;j++){
            if (map_flag[i][j] == 0 and mapp[i][j] == '1')
            	ans.push_back(bfs(i, j));
        }
    }
	cout << ans.size()<<endl;
    sort(ans.begin(),ans.end());
	for (auto i : ans)
		cout<<i<<endl;
    return 0;
}

File: level2/장애물인식프로그램.py
from collections import deque
N = int(input())

dy = [-1, 1, 0, 0]
dx = [0, 0, -1, 1]
ans = []
mapp = []
for i in range(N):
    line = list(map(int,input()))
    mapp.append(line)

checked = [[False] * N for _ in range(N)]

def bfs(x,y):
    cnt = 1
    queue = deque()
    queue.append((x, y))
    checked[x][y] = True
    while queue:
        x,y = queue.popleft()
        for i in range(4):
            nx = x + dx[i]
            ny = y + dy[i]
            if -1 < nx < N and -1 < ny < N:
                if not checked[nx][ny] and mapp[nx][ny] == 1:
                    queue.append((nx, ny))
                    checked[nx][ny] = True
                    cnt +=1            
    ans.append(cnt)

for i in range(N):
    for j in range(N):
        if not checked[i][j] and mapp[i][j] == 1:
            bfs(i, j)

print(len(ans))
for blocks in sorted(ans):
    print(blocks)